source/contributing/howtogit.rst

```diff
@@ -0,0 +1,241 @@
+to develop and test different features or additions at the same time, without causing problems and errors due to
+interferences of the additions. It's strongly advised that you **don't** make your changes on the ``master`` branch; instead,
+create a new branch yourself (with a sensible name) and make the changes there.
+
+This implies that we need to create a ``branch`` first, so let's go! You can do this via your client (there
+should be a ``create branch`` button somewhere), or you can use the CLI with git:
+
+.. code-block:: none
+
+  git checkout -b [name_of_your_new_branch]
+
+This will create a ``branch`` with the name of your choice and switch to it. All changes you're about to make will be
+on this branch. If you need to switch to another branch ( for example ``master``), just reuse this command. Third step
+done! Good job so far! To get an overview of your branches, just have a look at your git client or use:
+
+.. code-block:: none
+
+  git branch
+
+.. image:: /images/contributing/repo-branch.svg
+    :alt: Repo forking
+
+**Now it's time to make your changes**. Use the editor or IDE of your choice to do this.
+
+4. Test if Your Changes Work
+----------------------------
+
+For SpongeAPI and the implementations you have to run ``gradle compileJava``. Proceed to the next step if it finishes
+without errors. If it doesn't, make the appropriate corrections and try again.
+
+For SpongeDocs you can just submit your PR. It will get built automatically and reveal possible errors. Another option
+is to build the Docs locally. Have a look at the
+`Readme.md on the Docs <https://github.com/SpongePowered/SpongeDocs/blob/master/README.md>`_ for further instructions.
+
+5. Commit the Changes
+---------------------
+
+When you're done, you need to bundle them into a single package (a ``commit``) and get them into the branch. Again your
+git client will help you out. Add a meaningful name to your commit and a short description if needed. This can be done
+via CLI too:
+
+First collect all files and folders you want to put into a commit:
+
+.. code-block:: none
+
+  git add <file>
+  git add <folder>
+
+Now that the files are added to your list of changes you want included in the commit, just do
+
+.. code-block:: none
+
+  git commit
+
+It will open a text window, where you can add a message if you desire.
+
+.. note::
+  You can have multiple commits in a PR. Just go ahead and change everything you need and commit the changes.
+  You can merge the commits onto a single commit later.
+
+So now, the sixth step is done. Almost there!
+
+.. image:: /images/contributing/repo-commit.svg
+    :alt: Repo commit
+
+6. Sync to GitHub
+-----------------
+
+Now we need to get the changes to your fork on GitHub. Everything you've made so far is only stored locally
+right now. As always, you can use your git client to do this (there's a button somewhere), or you can do it via CLI:
+
+.. code-block:: none
+
+  git push
```

Comment:
I'd be a bit more specific here - best to use the full command: `git push <remote> <branch>`